src/IrcConnection/IrcConnection.cs

```diff
@@ -819,6 +819,9 @@ public void Disconnect()
             _IsConnected = false;
             _IsRegistered = false;
             
+            // signal ReadLine() to check IsConnected state
+            _ReadThread.QueuedEvent.Set();
```

Comment:
Isn't ReadLine() returning if the socket/stream underneath was closed? I only know of the issue (#4) with Thread.Abort() that needs a trick in the socket to bail out if it is still inside a blocked socket read. Or _is_ this the fix of #4?